### DISCUSSION TITLE
How to find package information for an older version?

### DISCUSSION THREAD

**Discussion by nick1989Gr** [SwiftPackageIndex/SwiftPackageIndex-Server#3520] [Q&A] [answered]
I would like to use https://swiftpackageindex.com in order to see package information for an older version. For example when I want to look on the data for `Moya/Moya` I hit `https://swiftpackageindex.com/Moya/Moya`. Currently the latest version is `15.0.3`. What if I want to check the package information for `15.0.2`. Is there a way to do this?

**Comment by daveverwer** [comment]
Hi @nick1989Gr 

I'm sorry but we do not keep old package metadata. We do keep documentation for older versions, but only if they host documentation with us (which Moya does not) but we don't keep any of the old compatibility or other metadata.

Is it compatibility information you were interested in? If not, what is your use case for the older data?

**Reply by nick1989Gr** [reply]
Thank you for the quick response. I am mainly interested in License information depending on the version. Sometimes package licenses change from version to version. So ideally I would like to be able to check the license of a package on a certain version.

**Reply by finestructure** [reply]
This is pretty straightforward on Github directly - would that work for you? For example, here's the full licensing history: https://github.com/Moya/Moya/commits/master/License.md

Or you can view the file at certain revisions via the version picker:

<img width="1349" alt="Screenshot 2024-11-25 at 12 43 26" src="https://github.com/user-attachments/assets/ccc8e82c-f402-4c40-929b-b6c8f01f5597">

**Reply by nick1989Gr** [reply]
@finestructure Thanks for your suggestions. Indeed these are valid ways of finding out if a package license has changed. Ideally I was looking for an API to get data in the form of `[version, license name]`. I already know that `swiftpackageindex` does not provide an API yet so I was trying to figure out if there is a way to get that data from  https://swiftpackageindex.com/. For example I can scrape the License of `Maya/Maya` from `https://swiftpackageindex.com/maya/maya`. Ideally I would like to do that also for older versions. 

I guess I can achieve the same by getting all the releases from a Swift github repo, fetching the `License` file and then using a license classification algorithm. I was hoping there is an easier way to do this.

**Reply by finestructure** [reply, marked as answer]
If you're looking for an API, you could use the Github REST API directly. The [license endpoint](https://docs.github.com/en/rest/licenses/licenses?apiVersion=2022-11-28) has a `ref` parameter to query by version. We're doing the same thing more or less, you might as well cut out the middle man 🙂
